modules/vpc-endpoints/main.tf

```diff
@@ -22,7 +22,7 @@ resource "aws_vpc_endpoint" "this" {
   for_each = local.endpoints
 
   vpc_id            = var.vpc_id
-  service_name      = data.aws_vpc_endpoint_service.this[each.key].service_name
+  service_name      = lookup(each.value, "service_name", null) != null ? lookup(each.value, "service_name", null) : data.aws_vpc_endpoint_service.this[each.key].service_name
```

Comment:
```suggestion
  service_name      = try(each.value.service_name, data.aws_vpc_endpoint_service.this[each.key].service_name)
```

Reply:
It might work if no data source is filtered then I'm going to test it then apply it.